packages/x-data-grid/src/hooks/features/virtualization/useGridVirtualScroller.tsx

```diff
@@ -216,9 +216,14 @@ export const useGridVirtualScroller = () => {
   );
 
   const triggerUpdateRenderContext = useEventCallback(() => {
+    const scroller = scrollerRef.current;
+    if (!scroller) {
```

Comment:
I still can't help myself asking if this is not hiding something. It could be about `useTimeout` that uses `useOnMount`, that uses `useEffect` so that call the clear function async, so after the DOM node was unmounted, ref was set to null.

Reply:
> after the DOM node was unmounted, ref was set to null

I tried reproducing it https://github.com/mui/mui-x/issues/14987#issuecomment-2489631711, but I couldn't get any scrolling callbacks called after the grid was unmounted.